File: app/src/main/java/com/example/socialapp/data/repositories/PostsRepository.java
package com.example.socialapp.data.repositories;

import androidx.lifecycle.LiveData;

import com.example.socialapp.models.Posts;
import com.example.socialapp.network.livedataadapter.ApiResponse;

public interface PostsRepository {
    LiveData<ApiResponse<Posts>> getPosts(String fields, String accessToken);
}

File: app/src/main/java/com/example/socialapp/data/repositories/PostsRepository.kt
package com.example.socialapp.data.repositories

import androidx.lifecycle.LiveData
import com.example.socialapp.models.Posts
import com.example.socialapp.network.livedataadapter.ApiResponse

interface PostsRepository {

    fun getPosts(fields: String, accessToken: String): LiveData<ApiResponse<Posts>>
}
